### PR TITLE
lib-user: Add BarChart component

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-user

## Describe your changes
- add BarChart component, with related mock data, stories, and tests

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What user actions should my reviewer step through to review this PR?
  - review [related Figma designs](https://www.figma.com/file/qbqbmR3t5XV6eKcpuRj7mG/Group-Stats?type=design&node-id=96-831&mode=design&t=jPCpzA45eJxYfgRx-0)
  - review storybooks
- Which storybook stories should be reviewed?
  - lots! starting with http://localhost:6008/?path=/story/components-shared-barchart--default
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes...
  - add gradient to bars
  - x-axis label granularity - https://github.com/zooniverse/front-end-monorepo/issues/5841
  - accessibility enhancement for bars / chart data - https://github.com/zooniverse/front-end-monorepo/issues/5842

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

File: packages/lib-user/src/components/shared/BarChart/BarChart.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { DataChart, Text } from 'grommet'
+import { arrayOf, number, shape, string } from 'prop-types'
+import withResponsiveContext from '@zooniverse/react-components/helpers/withResponsiveContext'
+
+import getDateRangeLabel from './helpers/getDateRangeLabel'
+
+function BarChart ({
+  data = [],
+  dateRange = 'Last7Days',
```
Comment (goplayoutside3) [inline packages/lib-user/src/components/shared/BarChart/BarChart.js:9]: Suggestion to create a map similar to `asyncStates` where you define the string options once, and then access the  `dateRange` map from anywhere in lib-user. `dateRange` will likely be passed to BarChart from a parent that contains the range options dropdown, and all of the range options are already used in BarChart's story too. The prop initialization here would look something like `dateRange.Last7Days`. Another example of this is [subjectViewers.js](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/helpers/subjectViewers/subjectViewers.js).

**Review by goplayoutside3 — COMMENTED** [review]
This is looking good! Just a couple of things to double check:
1. Are you still working toward medium granularity? I saw the note in the PR description, but that the PR is ready for review so went ahead and ran locally.
2. I noticed there's an intentional design choice to not include axis labels. This makes sense when viewing the BarChart with the context of range selected via dropdown menus, but it did make me pause and think about accessibility for a minute. I like that via keyboard you can focus on the chart and use the arrow keys to move through each bar, but was confused about what will happen if someone is using a screen reader. I tried VoiceOver, and when focused on the chart it tells me I'm looking at a group, but no spoken details about the data. Is that something you've had luck with testing?

File: packages/lib-user/src/components/shared/BarChart/BarChart.js (hunk 1 above)
Comment (mcbouslog) [reply packages/lib-user/src/components/shared/BarChart/BarChart.js]: Yes, great suggestion, will do.

**Comment by mcbouslog** [comment]
1. I've opened a Grommet issue related to the medium granularity issue - [grommet/issues/7085](https://github.com/grommet/grommet/issues/7085)
2. Yes great point. The keyboard access to each bar's detail is good, but it's missing the screen reader (VoiceOver at least) announcement. I think this is another Grommet issue. I'll confirm in the Grommet Slack and follow-up accordingly. It might be as straightforward as adding `a11yTitle` to the Text components in [these lines of code](https://github.com/grommet/grommet/blob/master/src/js/components/DataChart/Detail.js#L194-L195) 🤞 .

**Comment by mcbouslog** [comment]
Noting [the YourStats component](https://zooniverse.github.io/front-end-monorepo/@zooniverse/fe-project/index.html?path=/story/project-app-screens-classify-your-stats--your-stats) (built with visx) works well with a screen reader (and generally otherwise) as a potential alternative to the Grommet DataChart component.

**Comment by mcbouslog** [comment]
Updated initial comment with links to open issues and added general chart aria-label; I think ready for another review.

**Review by goplayoutside3 — APPROVED** [review]
Looks good!

**Pull request merged**